### DISCUSSION TITLE
Question about the logic for Private Routes in Nextjs

### DISCUSSION THREAD

**Discussion by erikmaddy** [nhost/nhost#141] [Q&A]
I have this working now but I wanted to see if anyone could tell me why one way works and another doesn't?

I was trying to put the logic for the Private Routes directly in the _app.js file instead of putting it in a separate component and then wrapping the Private Routes with the RequireAuth Component. When I would sign in, I would get a session back but there wouldn't be a re-render and I would just stay on the sign in page.

Once I switched it to a separate component, everything worked as expected. 

I have the working example and the broken example in the _app.js file.
[https://github.com/CovenantChurch/nhost-auth-example](https://github.com/CovenantChurch/nhost-auth-example)

I appreciate any insight you can give.

**Comment by st3phan** [comment]
Correct me if I'm wrong but thats probably because `_app.jsx` is not re-rendering but everything in `Component` is. That how Next works.

**Reply by sangdth** [reply]
I think this is how React works in general.

**Comment by plmercereau** [comment]
Hello, please have a look at the documentation of our new [@nhost/nextjs package](https://docs.nhost.io/reference/nextjs)
